test/plugin/base_test.rb

```diff
@@ -81,26 +81,26 @@ def test_configure_custom_metadata
     assert_equal CUSTOM_VM_ID, d.instance.vm_id
   end
 
-  def test_configure_invalid_metadata_missing_parts
+  def test_configure_metadata_missing_parts_on_other_platforms
     setup_no_metadata_service_stubs
     Fluent::GoogleCloudOutput::CredentialsInfo.stubs(:project_id).returns(nil)
-    { CONFIG_MISSING_METADATA_PROJECT_ID => ['project_id'],
-      CONFIG_MISSING_METADATA_ZONE => ['zone'],
-      CONFIG_MISSING_METADATA_VM_ID => ['vm_id'],
-      CONFIG_MISSING_METADATA_ALL => %w(project_id zone vm_id)
-    }.each_with_index do |(config, parts), index|
-      exception_count = 0
+    [[CONFIG_MISSING_METADATA_PROJECT_ID, ['project_id'], false],
+     [CONFIG_MISSING_METADATA_ZONE, [], true],
+     [CONFIG_MISSING_METADATA_VM_ID, [], true],
+     [CONFIG_MISSING_METADATA_ALL, ['project_id'], false]
+    ].each_with_index do |(config, missing_parts, is_valid_config), index|
       begin
         create_driver(config)
+
+        assert_true is_valid_config, "Index #{index} failed."
```

Comment:
The message should probably be more descriptive, e.g.: `"Invalid config at index #{index} should have raised an error."`. Same below, e.g.: `"Valid config at index #{index} should not have raised an error #{error}."`.